### PR TITLE
Release/v0.2.0

### PR DESCRIPTION
**Pull request by michaloo**
- BREAKING - chaning the name of segments filter (`segment_filter` -> `synchronized_segments`)
- add prefixing selected attributes with `traits_` to enhance matching (sometimes the event comes with the prefix while the attribute key in settings it without it)
- introduce segments based filtering on the `/batch` endpoint
- enhance the documentation in readme and in settings UI
- cleanup the code
- replace `utils` with `hull-ship-base` prototype

### CODE REVIEW

**Comment by michaloo** [comment]
[Prepare Webhooks Ship for Release](https://trello.com/c/OrJeCD3H/150-prepare-webhooks-ship-for-release)

File: assets/readme.md [hunk 1]
```diff
@@ -9,15 +9,15 @@ This Ship sends user updates as Webhooks
 
 ## Configuring the Webhooks connector
 
-There are two sections in Hull's Settings tab to help you define who will be sent as POST webhooks
+There are two sections in Hull's Settings tab to help you define who will be sent as POST webhooks.
```
Comment (unity) [inline assets/readme.md]: who => which users

File: assets/readme.md [hunk 2]
```diff
@@ -9,15 +9,15 @@ This Ship sends user updates as Webhooks
...
 
 The first section has a Segments filter.
 
-It defines who will be sent. An empty list sends no one. To start, create a User segment defining who should be sent.
+It defines who will be sent. An empty list sends no one. To start, create a User segment defining who should be sent. This is a global filter, the second condition would be checked only if the User matches this filter.
```
Comment (unity) [inline assets/readme.md]: the second condition would be checked => The conditions below will only be checked

File: assets/readme.md [hunk 3]
```diff
@@ -9,15 +9,15 @@ This Ship sends user updates as Webhooks
...
 
 The Second section defines additional conditions to send a user. User will be sent as soon as one of these conditions match.
 
-- When entering and/or leaving a given segment
+- When entering and/or leaving a given segment - NOTE: the global segment filter checks user segment information after the segment change event
```
Comment (unity) [inline assets/readme.md]: Confusing. Not sure it clarifies anything.
Pretty hard to explain here... Will think of something, let's not add it for now

File: manifest.json [hunk 4]
```diff
@@ -34,6 +34,7 @@
     {
       "name": "webhooks_segments",
       "title": "Send User when Segments changes",
+      "description": "NOTE: the global segment filter checks user segment information after the segment change event",
```
Comment (unity) [inline manifest.json]: confusing. let's not add it for now

**Pull request merged**